client/dashboard/src/pages/sources/SourceDetails.tsx

```diff
@@ -0,0 +1,387 @@
+              </div>
+              <div>
+                <Type className="text-sm text-muted-foreground mb-1">Type</Type>
+                <Type className="font-medium">{sourceType}</Type>
+              </div>
+              {!isOpenAPI && source && "runtime" in source && (
+                <div>
+                  <Type className="text-sm text-muted-foreground mb-1">
+                    Runtime
+                  </Type>
+                  <Type className="font-mono text-sm">{source.runtime}</Type>
+                </div>
+              )}
+              <div>
+                <Type className="text-sm text-muted-foreground mb-1">
+                  Last Updated
+                </Type>
+                <div className="flex items-center gap-2">
+                  <Calendar className="h-4 w-4 text-muted-foreground" />
+                  <Type className="text-sm">{lastUpdated}</Type>
+                </div>
+              </div>
+            </div>
+          </div>
+
+          {/* Toolsets Using This Source */}
+          <div className="rounded-lg border bg-card p-6">
+            <Type as="h2" className="text-lg font-semibold mb-4">
+              Used in Toolsets ({toolsetsUsingSource.length})
+            </Type>
+            {toolsetsUsingSource.length === 0 ? (
+              <div className="text-center py-8 text-muted-foreground">
+                <Package className="h-12 w-12 mx-auto mb-3 opacity-50" />
+                <Type>This source is not currently used in any toolsets</Type>
+              </div>
+            ) : (
+              <div className="space-y-2">
+                {toolsetsUsingSource.map((toolset) => (
+                  <div
+                    key={toolset.slug}
+                    onClick={() => routes.toolsets.toolset.goTo(toolset.slug)}
+                    className="flex items-center justify-between p-3 rounded-md border bg-surface-secondary hover:bg-surface-tertiary transition-colors cursor-pointer"
+                  >
+                    <div className="flex-1">
+                      <Type className="font-medium">{toolset.name}</Type>
+                      {toolset.description && (
+                        <Type className="text-sm text-muted-foreground mt-1">
+                          {toolset.description}
+                        </Type>
+                      )}
+                    </div>
+                    <Badge variant="outline" className="ml-2">
+                      {
+                        toolset.toolUrns?.filter((urn) =>
+                          sourceToolUrns.has(urn),
+                        ).length
+                      }{" "}
+                      tools
+                    </Badge>
+                  </div>
+                ))}
+              </div>
+            )}
+          </div>
+        </div>
+
+        {/* View Spec/Source Modal */}
+        <Dialog open={isModalOpen} onOpenChange={setIsModalOpen}>
+          <Dialog.Content className="min-w-[80vw] h-[90vh]">
```

Comment:
Can we make this its own component in `components/sources/ViewSourceDialogContent.tsx`? Probably also want to unify it with the dialog that renders on the toolset page "More Actions" flow but understand if we don't want to scope that in